clients/drcachesim/tests/record_filter_unit_tests.cpp

```diff
@@ -1356,6 +1356,22 @@ test_null_filter()
         FATAL_ERROR("Failed to run record filter: %s",
                     record_analyzer.get_error_string().c_str());
     }
+    if (!record_analyzer.print_stats()) {
+        FATAL_ERROR("Failed to print record filter stats: %s",
+                    record_analyzer.get_error_string().c_str());
+    }
+
+    // Ensure schedule files were written out.  We leave validating their contents
+    // to the end-to-end tests which run invariant_checker.
+    std::string serial_path = output_dir + DIRSEP + DRMEMTRACE_SERIAL_SCHEDULE_FILENAME;
+#ifdef HAS_ZLIB
+    serial_path += ".gz";
+#endif
+    CHECK(dr_file_exists(serial_path.c_str()), "Serial schedule file missing\n");
```

Comment:
Done.  This revealed that the destructor hadn't been called to flush the file: so I added a local scope; and since I used dr_open_file to get the size I added dr_standalone_init to avoid dr_close_file complaining.